includes/utilities.inc

```diff
@@ -48,3 +76,66 @@ EOQ;
 
   return $associations;
 }
+
+/**
+ * Get the associated object for the given EAD object and refid.
+ *
+ * @param string $eadid
+ *   The PID of the EAD object to retrieve the association of.
+ * @param string $refid
+ *   The level ID to retrieve the association of.
+ *
+ * @return string
+ *   The PID of the associated object or NULL.
+ */
+function islandora_ead_retrieve_refid_association($eadid, $refid) {
+  $tuque = islandora_get_tuque_connection();
+  $sparql_query = <<<EOQ
+SELECT ?partid
+FROM <#ri>
+WHERE {
+  ?partid <fedora-rels-ext:isPartOf> <info:fedora/$eadid> ;
+          <fedora-rels-ext:eadLevelId> <info:fedora/$refid> .
+}
+EOQ;
+
+  $sparql_results = $tuque->repository->ri->sparqlQuery($sparql_query);
+  return $sparql_results ? $sparql_results[0]['partid']['value'] : NULL;
+}
+
+/**
+ * Get the level information for the given EAD object and refid.
+ *
+ * @param AbstractObject $ead
+ *   The EAD object to retrieve the association of.
+ * @param string $refid
+ *   The level ID to retrieve the association of.
+ *
+ * @return string
+ *   String describing the level information.
+ */
+function islandora_ead_get_level_info(AbstractObject $ead, $refid) {
+  $xsl = new DOMDocument();
+  $xsl->load(drupal_get_path('module', 'islandora_ead') . '/xml/getEadLevels.xsl');
+
+  $xslt = new XSLTProcessor();
+  $xslt->importStyleSheet($xsl);
+
+  $ead_dom = new DOMDocument();
+  $ead_dom->loadXML($ead['EAD']->content);
+  $ead_dom = $xslt->transformToDoc($ead_dom);
+
+  $xpath = new DOMXPath($ead_dom);
+  $entity = $xpath->query("/entities/entity[id = '$refid']")->item(0);
+  $containers = array();
+  foreach ($xpath->query('container', $entity) as $container) {
+    $type = $xpath->query('type', $container)->item(0)->textContent;
```

Comment:
Could do this with `->evaluate('string(whatever)', $container)`. Also any concern with these values being empty?